src/pudl/transform/ferc1.py

```diff
@@ -6322,6 +6266,100 @@ def table_to_column_to_check() -> dict[str, list[str]]:
     }
 
 
+def remove_rare_utility_type_subdimensions_rows(df: pd.DataFrame) -> pd.DataFrame:
+    """Remove the rare, non-total utility types when all values are duplicated.
+
+    We remove the records of non-total utility_type's for xbrl_factoid's when almost
+    all instances of that xbrl_factoid show up with just a utility_type of "total". We
+    can only do this confidently because we also check that all of the dollar_value in
+    those records are exactly the same as the corresponding records with utility_type
+    of "total" or the non-total utility_type's have null dollar_value's.
+
+    This data isn't incorrect, it just interferes with how we process the calculations
+    embedded within these tables. This is why we are applying this within
+    :func:`_core_ferc1__table_dimensions` because that is what we use to build the
+    calculation components table.
+    """
+    # made these variables so we could generalize to other tables if we desired that.
+    xbrl_factoid = "xbrl_factoid"
+    dimension_col = "utility_type"
+    idx = ["utility_id_ferc1", "report_year", xbrl_factoid]
+
+    # first we need to find the xbrl_factoid's where there are mostly only totals
+    # within the dimension_col. Which is to say there are a few rare non-total
+    # utility_type's.
+    tot_mask = df[dimension_col] == "total"
+    # build a dataframe of xbrl_factoid's with a column for the count of the records
+    # with utility_type total and another column with the count for all the other
+    # utility_type's
+    util_type_count = pd.merge(
```

Comment:
If we do generalize to other dimensions in other tables, we'll probably want to make `util_type_count` more generic to go along with `dimension_col`.

Reply:
yea i didn't bc the more i preemptively generalize the more unreadable it seems to me... but i can

Reply:
I don't think it needs to be swapped now necessarily just noticed that you'd generalized some parts and not others.